src/sentry/replays/usecases/query/__init__.py

```diff
@@ -163,6 +165,20 @@ def query_using_optimized_search(
             SearchFilter(SearchKey("environment"), "IN", SearchValue(environments)),
         ]
 
+    for i in range(len(search_filters)):
+        sf = search_filters[i]
+        if sf.key.name in VIEWED_BY_ME_KEY_ALIASES:
+            # "viewed_by_me" is not a valid query field.
+            # It's a convenience alias for users without the admin access to lookup ids.
+            if sf.operator not in ["=", "!="]:
+                raise ParseError(f"Invalid operator specified for {sf.key.name}")
+            is_me = not (
+                (sf.operator == "=") ^ sf.value.value
+            )  # True if both expressions are the same, else False.
+            search_filters[i] = SearchFilter(
+                SearchKey("viewed_by_id"), "=" if is_me else "!=", SearchValue(request_user_id)
```

Comment:
Let's assign this inline `if` to a variable `operator`.  A little more verbose but it will lower the density of this code.